firebase-vscode/src/data-connect/terminal.ts

```diff
@@ -99,16 +98,25 @@ export function registerTerminalTasks(
     if (settings.exportOnExit) {
       cmd += ` --export-on-exit ${settings.exportPath}`;
     }
+    vscode.window.showInformationMessage("Starting emulators... Please see terminal.");
     runTerminalTask(
       "firebase emulators",
       cmd,
       { focus: true },
     );
+  };
+  const startEmulatorsTaskBroker = broker.on("runStartEmulators", () => {
```

Comment:
I saw `broker.on("runStartEmulators")` twice in the code base.

Here and [in emulator.ts](https://github.com/firebase/firebase-tools/blob/77620c041c76a2c6bb4f218e2dc70ab4f845adae/firebase-vscode/src/core/emulators.ts#L24-L26). Would it make sense to dedup them?

Move to `emulator.ts` and re-use `startEmulators` method?